tests/e2e/run.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+# This script install dependencies of service side of e2e tests and runs e2e test
+
+# ./run.sh <RTOS connectivity interface>
+# e.g. ./run.sh veth1
+
+if [[ ! -z ${DEBUG_SHELL} ]]
+then
+  set -x # Activate the expand mode if DEBUG is anything but empty.
+fi
+
+set -o errexit # Exit if command failed.
+set -o nounset # Exit if variable not set.
+set -o pipefail # Exit if pipe failed.
+
+die() {
+    echo "${1:-"Unknown Error"}" 1>&2
+    exit 1
+}
+
+abspath () {
+    case "$1" in /*)printf "%s\n" "$1";; *)printf "%s\n" "$PWD/$1";; esac;
+}
+
+[ -n $1 ] || die "ethernet interface need to be passed; like veth1"
+
+filepath=`abspath $0`
+dir=`dirname $filepath`
+
+echo -e "Updating FreeRTOSConfig.h to point to interface $1"
+index=`tcpdump --list-interfaces | grep -Ei "([0-9]+).$1" | sed -E 's/^([0-9]+).*/\1/g'`
+sed -i "s/#define configNETWORK_INTERFACE_TO_USE.*/#define configNETWORK_INTERFACE_TO_USE ($index)/g" $dir/device/FreeRTOSConfig.h
```

Comment:
Is there a way we could not use `sed`.  Like any way under the sun:)?

Once we get cmake support, the lines below where you're doing `make` presumably that gets turned to a cmake 1st, right?  Could we use a cmake define "thingy" to enable this?

Reply:
Index of interface is dynamic and will change based on which VM it is running so need to find index based on machine.

Yes make will be replaced via cmake.